server/src/main/java/org/elasticsearch/index/query/RangeQueryBuilder.java

```diff
@@ -433,33 +433,30 @@ public String getWriteableName() {
 
     // Overridable for testing only
     protected MappedFieldType.Relation getRelation(QueryRewriteContext queryRewriteContext) throws IOException {
-        CoordinatorRewriteContext coordinatorRewriteContext = queryRewriteContext.convertToCoordinatorRewriteContext();
-        if (coordinatorRewriteContext != null) {
-            final MappedFieldType fieldType = coordinatorRewriteContext.getFieldType(fieldName);
-            if (fieldType instanceof final DateFieldMapper.DateFieldType dateFieldType) {
-                if (coordinatorRewriteContext.hasTimestampData() == false) {
-                    return MappedFieldType.Relation.DISJOINT;
-                }
-                long minTimestamp = coordinatorRewriteContext.getMinTimestamp();
-                long maxTimestamp = coordinatorRewriteContext.getMaxTimestamp();
-                DateMathParser dateMathParser = getForceDateParser();
-                return dateFieldType.isFieldWithinQuery(
-                    minTimestamp,
-                    maxTimestamp,
-                    from,
-                    to,
-                    includeLower,
-                    includeUpper,
-                    timeZone,
-                    dateMathParser,
-                    queryRewriteContext
-                );
+        final MappedFieldType fieldType = queryRewriteContext.getFieldType(fieldName);
+        if (queryRewriteContext instanceof final CoordinatorRewriteContext coordinatorRewriteContext
```

Comment:
maybe the convertToCoordinatorRewriteContext was better in that it let us avoid this instanceof check?

Reply:
Yes I was considering a few more small changes before asking for a review...I wanted to run the full CI on it including full bwc (even if I don't expect bwc issues). Thanks.

Reply:
I also need to go through the JavaDoc to make sure I update everything to reflect the refactoring.

Reply:
thanks! ping me for review when ready ;)